app/javascript/controllers/sidebar_group_controller.js

```diff
@@ -16,4 +22,63 @@ export default class extends Controller {
       }
     })
   }
+
+  // For group list with anchor links where toggleShow() does not work because
+  // active class does not get triggered in sidebar_helper.rb
+  toggleShowAnchorList () {
+    this.titleTarget.classList.remove('collapsed')
+    this.listTarget.classList.add('show')
+  }
+
+  // Highlights menu items as user scrolls on page. Implemented for casa_org/1/edit#organization-details.
+  menuHighlight () {
+    /**
+     * Constructs a map where the keys are header IDs extracted from the href attributes of the links,
+     * and the values are the corresponding link elements.
+     *
+     * @returns {Object} A hash map with header IDs as keys and link elements as values.
+     */
+    const linkTargetsMap = () => {
+      const hash = {}
+      this.linkTargets.forEach(link => {
+        const href = link.children[0].href
+        const headerID = href.substring(href.indexOf('#') + 1)
+        hash[headerID] = link
+      })
+      return hash
+    }
+
+    // Call linkTargetsMap() and assign it to linkHash
+    const linkHash = linkTargetsMap()
+
+    // add class active to link if it on page
+    const observer = new IntersectionObserver(entries => {
+      entries.forEach(entry => {
+        if (entry.intersectionRatio > 0) {
+          this.linkTargets.forEach(link => link.classList.remove('active'))
+
+          const headerID = entry.target.id
+          const activeLink = linkHash[headerID]
```

Comment:
I think this will follow current coding convention!

```suggestion
          const headerId = entry.target.id
          const activeLink = linkHash[headerId]
```